pynxtools/nexus/nexus.py

```diff
@@ -574,6 +583,72 @@ def get_node_concept_path(elem):
     return str(elem.get('nxdlbase').split('/')[-1] + ":" + elem.get('nxdlpath'))
 
 
+def get_nxdl_attr_doc(  # pylint: disable=too-many-arguments,too-many-locals
+        elem, elist, attr, hdf_node, logger, doc, nxdl_path, req_str, path, hdf_info):
+    """Get nxdl documentation for an attribute"""
+    new_elem = []
+    old_elem = elem
+    for elem_index, act_elem1 in enumerate(elist):
+        act_elem = act_elem1
+        # NX_class is a compulsory attribute for groups in a nexus file
+        # which should match the type of the corresponding NXDL element
+        if attr == 'NX_class' and not isinstance(hdf_node, h5py.Dataset) and elem_index == 0:
+            elem = None
+            logger, doc, attr = write_doc_string(logger, doc, attr)
+            new_elem = elem
+            break
+        # units category is a compulsory attribute for any fields
+        if attr == 'units' and isinstance(hdf_node, h5py.Dataset):
+            req_str = "<<REQUIRED>>"
```

Comment:
I am wondering if it is possible to use `get_required_string()` instead of hard coding <<REQUIRED>>.

Reply:
Curently, get_required_string(nxdl_elem) only returns requirements defined in the nxdl definition files. On he other hand the above requirements come from NeXus conventions which are not written in the nxdl files.